cmd/erasure-sets.go

```diff
@@ -92,6 +92,12 @@ type erasureSets struct {
 	lastConnectDisksOpTime time.Time
 }
 
+func (s *erasureSets) getErasureDisks() [][]StorageAPI {
+	s.erasureDisksMu.RLock()
```

Comment:
You can just use s.getDisks()

Reply:
I find out 
```go
// GetDisks returns a closure for a given set, which provides list of disks per set.
func (s *erasureSets) GetDisks(setIndex int) func() []StorageAPI {
	return func() []StorageAPI {
		s.erasureDisksMu.RLock()
		defer s.erasureDisksMu.RUnlock()
		disks := make([]StorageAPI, s.setDriveCount)
		copy(disks, s.erasureDisks[setIndex])
		return disks
	}
}
```
@harshavardhana . Use this?

Reply:
Yes this is exactly what you want